quickjs/qjs.c

```diff
@@ -311,50 +313,65 @@ int main(int argc, const char **argv) {
     CHECK2(ctx != NULL, -1);
     /* loader for ES6 modules */
     JS_SetModuleLoaderFunc(rt, NULL, js_module_loader, NULL);
-    js_std_add_helpers(ctx, argc - optind, argv + optind);
+    js_std_add_helpers(ctx, argc - optind, &argv[optind]);
     err = js_init_module_ckb(ctx);
     CHECK(err);
 
-    switch (type) {
-        case RunJsWithCode:
-            err = eval_buf(ctx, argv[1], strlen(argv[1]), "<cmdline>", 0);
-            if (err == 0) {
-                js_std_loop(ctx);
-            }
-            break;
-        case RunJsWithFile:
-            err = run_from_cell_data(ctx, false);
-            if (err == 0) {
-                js_std_loop(ctx);
-            }
-            break;
-        case RunJsWithFileSystem:
-            err = run_from_cell_data(ctx, true);
-            if (err == 0) {
-                js_std_loop(ctx);
-            }
-            break;
-        case RunJsWithDbgFile:
-            err = run_from_local_file(ctx, false);
-            if (err == 0) {
-                js_std_loop(ctx);
-            }
-            break;
-        case RunJsWithDbgFileSystem:
-            err = run_from_local_file(ctx, true);
-            if (err == 0) {
-                js_std_loop(ctx);
-            }
-            break;
-        case CompileWithFile:
-            JS_SetModuleLoaderFunc(rt, NULL, js_module_dummy_loader, NULL);
-            err = compile_from_file(ctx);
-            break;
-        default:
-            printf("unknow type: %d", type);
-            return -1;
+    if (cmdopt_has(co, "c")) {
+        JS_SetModuleLoaderFunc(rt, NULL, js_module_dummy_loader, NULL);
+        err = compile_from_file(ctx);
+        return err;
     }
-    CHECK(err);
+    if (cmdopt_get(co, "e")) {
+        const char *data = cmdopt_get(co, "e");
+        err = eval_buf(ctx, data, strlen(data), "<cmdline>", 0);
+        if (err == 0) {
+            js_std_loop(ctx);
```

Comment:
Can we move this call js_std_loop to later?

Reply:
js_std_loop is not needed when compiling bytecode, I prefer to put it in the branch where it is needed.

Reply:
I think, js_std_loop can be used while compiling.